### ISSUE TITLE
Can't turn off Debug logging to console.

### ISSUE THREAD

**Issue by maassql** [jmxtrans/jmxtrans#239]
My apologies if this is the incorrect forum for this type of problem/question.  I am new to the Java world.  I see other issues talking about other loggers being used in this project.  I suspect that's the case here, however, I haven't figured out how to set / administer them.

* I cloned straight out of GitHub.  Created jars using the pom.xml.
* I've ensured that the logging level is set to warn as evidenced by the system property at the command line : "-Djmxtrans.log.level=warn".
* I've added a root logger to log4j.xml - and set it to warn
* I note that the console output starts before log4j initialization
* I note that the console output is not formatted as per log4j.xml
* I note that nothing is being written to the rolling file appender writing to the file jmxtrans.log, as defined in log4j.xml

if the log4j.xml file is inside of the jar file:
```bash
14:24:42.051 [main] DEBUG c.googlecode.jmxtrans.JmxTransformer - Scheduled job: storm-clusternode3:8002-1426861482049-7125349009 for server: Server [host=storm-clusternode3, port=8002, url=null, cronExpression=null, numQueryThreads=2]
log4j: System property is :null
log4j: Standard DocumentBuilderFactory search succeded.
log4j: DocumentBuilderFactory is: com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderFactoryImpl
log4j:WARN Continuable parsing error 57 and column 23
log4j:WARN The content of element type "log4j:configuration" must match "(renderer*,throwableRenderer?,appender*,plugin*,(category|logger)*,root?,(categoryFactory|loggerFactory)?)".
log4j: debug attribute= "null".
log4j: Ignoring debug attribute.
log4j: reset attribute= "false".
log4j: Threshold ="null".
log4j: Level value for root is  [warn].
log4j: root level set to WARN
log4j: Class name: [org.apache.log4j.ConsoleAppender]
log4j: Parsing layout of class: "org.apache.log4j.PatternLayout"
log4j: Setting property [conversionPattern] to [[%d{dd MMM yyyy HH:mm:ss}] %-6r %-5p (%C:%L) - %m
].
log4j: Adding appender named [STDOUT] to category [root].
log4j: Retreiving an instance of org.apache.log4j.Logger.
log4j: Setting [com.googlecode.jmxtrans.model.output.GraphiteWriter] additivity to [true].
log4j: Level value for com.googlecode.jmxtrans.model.output.GraphiteWriter is  [WARN].
log4j: com.googlecode.jmxtrans.model.output.GraphiteWriter level set to WARN
log4j: Retreiving an instance of org.apache.log4j.Logger.
log4j: Setting [com.googlecode.jmxtrans.model.output] additivity to [true].
log4j: Level value for com.googlecode.jmxtrans.model.output is  [WARN].
log4j: com.googlecode.jmxtrans.model.output level set to WARN
log4j: Retreiving an instance of org.apache.log4j.Logger.
log4j: Setting [com.googlecode.jmxtrans] additivity to [true].
log4j: Level value for com.googlecode.jmxtrans is  [WARN].
log4j: com.googlecode.jmxtrans level set to WARN
log4j: Retreiving an instance of org.apache.log4j.Logger.
log4j: Setting [org.quartz] additivity to [true].
log4j: Level value for org.quartz is  [warn].
log4j: org.quartz level set to WARN
log4j: Level value for root is  [error].
log4j: root level set to ERROR
log4j: Class name: [org.apache.log4j.AsyncAppender]
log4j: Setting property [bufferSize] to [1024].
log4j: Setting property [blocking] to [true].
log4j: Setting property [locationInfo] to [true].
log4j: Attaching appender named [fileAppender] to appender named [asyncAppender].
log4j: Class name: [org.apache.log4j.DailyRollingFileAppender]
log4j: Setting property [file] to [./jmxtrans.log].
log4j: Setting property [append] to [true].
log4j: Setting property [datePattern] to ['.'yyyyMMdd].
log4j: Parsing layout of class: "org.apache.log4j.PatternLayout"
log4j: Setting property [conversionPattern] to [[%d{dd MMM yyyy HH:mm:ss}] [%t] %-6r %-5p (%C:%L) - %m
].
log4j: setFile called: ./jmxtrans.log, true
log4j: setFile ended
log4j: Appender [fileAppender] to be rolled at midnight.
log4j: Adding appender named [asyncAppender] to category [root].
14:24:42.215 [ServerScheduler_Worker-2] DEBUG com.googlecode.jmxtrans.jmx.JmxUtils - ----- Creating 15 query threads
```

If I pull log4j.xml from the jar file:
```bash
14:11:01.757 [ServerScheduler_Worker-6] DEBUG org.quartz.core.JobRunShell - Calling execute on job ServerJob.storm-clusternode3:8002-1426860661757-3992594647
14:11:01.757 [ServerScheduler_Worker-6] DEBUG c.googlecode.jmxtrans.jobs.ServerJob - +++++ Started server job: Server [host=storm-clusternode3, port=8002, url=null, cronExpression=null, numQueryThreads=2]
log4j: Trying to find [log4j.xml] using context classloader sun.misc.Launcher$AppClassLoader@1b219665.
log4j: Trying to find [log4j.xml] using sun.misc.Launcher$AppClassLoader@1b219665 class loader.
log4j: Trying to find [log4j.xml] using ClassLoader.getSystemResource().
log4j: Trying to find [log4j.properties] using context classloader sun.misc.Launcher$AppClassLoader@1b219665.
log4j: Trying to find [log4j.properties] using sun.misc.Launcher$AppClassLoader@1b219665 class loader.
log4j: Trying to find [log4j.properties] using ClassLoader.getSystemResource().
log4j: Could not find resource: [null].
log4j:WARN No appenders could be found for logger (org.jboss.logging).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
14:11:01.858 [ServerScheduler_Worker-5] DEBUG com.googlecode.jmxtrans.jmx.JmxUtils - ----- Creating 160 query threads
14:11:01.858 [ServerScheduler_Worker-1] DEBUG com.googlecode.jmxtrans.jmx.JmxUtils - ----- Creating 228 query threads
```

**Comment by maassql** [comment]
After re-reading other issues here, 
* I deleted the class TimeBasedRollingKeyOutWriter and removed dependencies from pom.xml upon ch.quos.logback.*
* rebuilt

and now am only seeing warning messages in the console

**Comment by gehel** [comment]
Yep, that seems the right approach. I need to find some time to clean
this TimeBasedRollingKeyOutWriter,
but I'm afraid I'm a bit too busy these days.

Sorry for the inconvenience.

On 20 March 2015 at 15:54, Jeff Maass <notifications@github.com> wrote:

> After re-reading other issues here,
>
>    - I deleted the class TimeBasedRollingKeyOutWriter and removed
>    dependencies from pom.xml upon ch.quos.logback.*
>    - rebuilt
>
> and now am only seeing warning messages in the console
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jmxtrans/jmxtrans/issues/239#issuecomment-84036577>.
>



-- 
mobile : +41 76 573 32 40
skype : Guillaume.Lederrey

projects :
* http://rwanda.ledcom.ch/
* http://trock.ch/
others :
* http://kiva.org/

**Comment by gehel** [comment]
Closing this as OP seems to have resolved his issue and the underlying problem will be addressed in #215.
